server/controllers/notification.js

```diff
@@ -0,0 +1,176 @@
+        notification: {
+          _id: notification.id,
+          notifyType: notification.notifyType,
+          title: notification.title,
+          description: notification.description,
+          sender: {
+            userId: user.id,
+            name: sender.name,
+            photo: sender.photo,
+          },
+          receivers: notification.receivers,
+          readBy: notification.readBy,
+          createdAt: notification.createdAt,
+        },
+      },
+    });
+  } else {
+    res.status(400);
+    throw new Error("Invalid notification data");
+  }
+});
+
+exports.markNotificationRead = asyncHandler(async (req, res, next) => {
+  const {
+    user,
+    body: { id },
+  } = req;
+
+  const receiver = await User.findById(user.id);
+
+  if (!receiver) {
+    res.status(401);
+    throw new Error("Not authorized");
+  }
+
+  const query = {
+    _id: id,
+    receivers: {
+      $in: receiver.id,
+    },
+  };
+
+  const data = {
+    $set: {
+      readBy: {
+        receiverId: receiver.id,
+      },
+    },
+  };
+
+  const notification = await Notification.findOneAndUpdate(query, data, {
+    new: true,
+  })
+    .populate({
+      path: "sender",
+      select: "userId name photo",
+    })
+    .select("-receivers -__v -updatedAt")
+    .exec();
+
+  if (notification) {
+    res.status(201).json({
+      success: {
+        notification,
+      },
+    });
+  } else {
+    res.status(400);
+    throw new Error("Invalid notification data");
+  }
+});
+
+exports.getAllNotifications = asyncHandler(async (req, res, next) => {
+  const {
+    query: {
+      limit = Infinity,
+      page = 1,
+      read = null,
+      sort = "createdAt",
+      order = 1,
+    },
+    user,
+  } = req;
+
+  const receiver = await User.findById(user.id);
+
+  if (!receiver) {
+    res.status(401);
+    throw new Error("Not authorized");
+  }
+
+  let query = {
+    receivers: {
+      $in: receiver.id,
+    },
+  };
+
+  const total = await Notification.countDocuments(query);
+
+  if (read === "true") {
+    query = {
+      receivers: {
+        $in: receiver.id,
+      },
+      readBy: {
+        $elemMatch: {
+          receiverId: receiver.id,
+        },
+      },
+    };
+  } else if (read === "false") {
+    query = {
+      receivers: {
+        $in: receiver.id,
+      },
+      readBy: {
+        $not: {
+          $elemMatch: {
+            receiverId: receiver.id,
+          },
+        },
+      },
+    };
+  }
+
+  const notifications = await Notification.find(query)
+    .skip((parseInt(page) - 1) * limit)
+    .limit(parseInt(limit))
+    .sort({ [sort]: parseInt(order) })
+    .populate({
+      path: "sender",
+      select: "userId name photo",
+    })
+    .select("-receivers -__v -updatedAt")
+    .exec();
+
+  if (notifications) {
+    res.status(200).json({
+      success: {
+        notifications,
+        total,
+      },
+    });
+  } else {
+    throw new Error("Unable to find any notifications for user");
+  }
```

Comment:
It makes more sense to return an empty list with a success code when there aren't any notifications found (more details [here](https://stackoverflow.com/a/13367198/15220729)).

You could make a case for an error when provided a page that's out of range though (e.g. there are only `5` pages but the client asks for `6`th page).